src/index.ts

```diff
@@ -283,37 +280,72 @@ export class Upload extends Pumpify {
   }
 
   private async startUploading() {
-    const reqOpts = {
-      method: 'PUT',
-      url: this.uri!,
-      headers: {
-        'Content-Range': 'bytes ' + this.offset + '-*/' + this.contentLength
-      }
-    };
+    // The buffer stream allows us to keep chunks in memory
+    // until we are sure we can successfully resume the upload.
+    const bufferStream = this.bufferStream || new PassThrough();
```

Comment:
Fixes #175. Went with lazy-creating it instead of using the constructor, just to be consistent with our optimization path.